homeworkW11D02/src/ba/bitcamp/homeworkW11D02/Server.java

```diff
@@ -0,0 +1,234 @@
+
+	// Window components
+	private JPanel mainPanel = new JPanel();
+	private JPanel downPanel = new JPanel();
+	private JPanel upPanel = new JPanel();
+	private JTextArea textArea = new JTextArea();
+	private JTextField textField = new JTextField();
+	private JScrollPane scrollPane = new JScrollPane(textArea,
+			JScrollPane.VERTICAL_SCROLLBAR_AS_NEEDED,
+			JScrollPane.HORIZONTAL_SCROLLBAR_AS_NEEDED);
+
+	// Server and regular socket
+	private ServerSocket server;
+	private Socket client;
+
+	// List of directories
+	private ArrayList<String> listOfDirectories;
+
+	/**
+	 * Constructor for creating Server chat
+	 */
+	public Server() {
+
+		/*
+		 * Adding KeyListener to text field
+		 */
+		textField.addKeyListener(new KeyAdapter() {
+			@Override
+			public void keyPressed(KeyEvent e) {
+
+				/*
+				 * If server user types ENTER key on the keyboard and if the
+				 * text field is not empty, written message will be sent to a
+				 * client
+				 */
+				if (e.getKeyCode() == KeyEvent.VK_ENTER
+						&& !textField.getText().equals("")) {
+					try {
+
+						BufferedWriter writer = new BufferedWriter(
+								new OutputStreamWriter(client.getOutputStream()));
+						writer.write("Server: " + textField.getText());
+						writer.newLine();
+						writer.flush();
+						textArea.append("Server: " + textField.getText() + "\n");
+						textArea.setCaretPosition(textArea.getDocument()
+								.getLength());
+						textField.setText("");
+					} catch (IOException e1) {
+						e1.printStackTrace();
+					}
+				}
+			}
+		});
+
+		/*
+		 * Configuring and adding components to lower panel
+		 */
+		downPanel.setBorder(BorderFactory.createTitledBorder("Write Message"));
+		downPanel.setLayout(new BorderLayout());
+		downPanel.add(textField, BorderLayout.CENTER);
+
+		/*
+		 * Configuring and adding components to upper panel
+		 */
+		upPanel.setBorder(BorderFactory.createTitledBorder("Chat room"));
+		upPanel.setLayout(new BorderLayout());
+		upPanel.add(scrollPane, BorderLayout.CENTER);
+
+		/*
+		 * Configuring text area
+		 */
+		textArea.setLineWrap(true);
+		textArea.setEditable(false);
+		textArea.setForeground(Color.BLUE);
+		textArea.setBackground(Color.ORANGE);
+		textArea.setFont(new Font("Serif", Font.BOLD, 15));
+
+		/*
+		 * Configuring and adding components to main panel
+		 */
+		mainPanel.setLayout(new BorderLayout());
+		mainPanel.add(upPanel, BorderLayout.CENTER);
+		mainPanel.add(downPanel, BorderLayout.SOUTH);
+
+		// Adding main panel to the window
+		add(mainPanel);
+
+		/*
+		 * Configuring window
+		 */
+		setTitle("Server chat");
+		setSize(450, 600);
+		setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+		setResizable(false);
+		setLocationRelativeTo(null);
+		setVisible(true);
+
+		/*
+		 * Starting server
+		 */
+		try {
+			server = new ServerSocket(6699);
+
+			// Waiting for client to connect
+			client = server.accept();
+
+		} catch (IOException e) {
+			e.printStackTrace();
```

Comment:
In case accept fails, you should exit since it means client was not created correctly.